src/common/perf_histogram.h

```diff
@@ -0,0 +1,213 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 OVH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_COMMON_PERF_HISTOGRAM_H
+#define CEPH_COMMON_PERF_HISTOGRAM_H
+
+#include "common/Formatter.h"
+#include "include/atomic.h"
+#include "include/int_types.h"
+
+#include <memory>
+
+class PerfHistogramCommon {
+public:
+  enum scale_type_d : uint8_t {
+    SCALE_LINEAR = 1,
+    SCALE_LOG2 = 2,
+  };
+
+  struct axis_config_d {
+    const char *m_name;
+    scale_type_d m_scale_type;
+    int64_t m_min;
+    int64_t m_quant_size;
+    int32_t m_buckets;
```

Comment:
@byo okay, i was trying to fix the issues reported by coverity, which complains that the fields of `axis_config_d` are not initialized. i know this is a false alarm now, but shall we set some safe values in `axis_config_d` ? like
```c++
  struct axis_config_d {
    const char *m_name = nullptr;
    scale_type_d m_scale_type = 0;
    int64_t m_min = 0;
    int64_t m_quant_size = 0;
    int32_t m_buckets =0;
}
```

what do you think?